awaitwhat/dot.py

```diff
@@ -36,7 +36,14 @@ def describe(task, current=None):
             name = task.get_name()
         except AttributeError:
             name = "Task"
-        # FIXME can we ever see a "done" task?
+        try:
+            done, pending = await asyncio.wait(task)
```

Comment:
This is very elegant, however, `await` is only allowed in an `async` function.
The current approach of `awaitwhat` library is that it takes a snapshot of tasks and coroutines, that is, we don't want to give the event loop a chance to run.

I think some other way to get this information is needed, something more low level.

Reply:
Understood!